### DISCUSSION TITLE
NuGet restore for .NET 6 Android project seems to ignore fallback frameworks

### DISCUSSION THREAD

**Discussion by leonluc-dev** [cake-build/cake#3674] [Q&A] [answered]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake .NET Tool

### Cake version

1.3.0

### Operating system

Windows

### Operating system architecture

64-Bit

### What are you seeing?

I have .NET 6 project targeting `net6.0-android` I want to build using cake. This works fine using a simple command line build like `dotnet build MyProject.csproj`, however when I build the same project in cake using `DotNetCoreBuild` I get the following error during the restore phase:

`NU1202: Package Xamarin.Google.Guava.ListenableFuture 1.0.0.2 is not compatible with net6.0-android31.0 (.NETCoreApp,Version=v6.0) / android-arm64. Package Xamarin.Google.Guava.ListenableFuture 1.0.0.2 supports: monoandroid50 (MonoAndroid,Version=v5.0)`

### What is expected?

The net6.0-android target supports monodroid packages and NuGet is supposed restore them appropriately.

### Steps to Reproduce

1. Create a .NET 6 project with target framework `net6.0-android`
2. Add a monodroid based NuGet package to it (for example Xamarin.Google.Guava.ListenableFuture 1.0.0.2).
    No other code should be required to reproduce the issue
3. Try to build it using the `DotNetCoreBuild` method in cake
4. The error mentioned above should occur

**Comment by augustoproiete** [comment, marked as answer]
Under the hood, Cake is calling `dotnet build` as well, so there shouldn't be any difference between you calling `dotnet build` directly or Cake calling it via `DotNetCoreBuild`.

If you run Cake with the argument `--settings_showprocesscommandline=true` you can see exactly the `dotnet build` command-line being executed, and that might give you some clues as to what's different.

e.g.

```powershell
dotnet cake --settings_showprocesscommandline=true
```

Otherwise, if you can put together a reproducible example, that would help troubleshoot this.

**Reply by leonluc-dev** [reply]
Thanks for the parameter hint. With that I was able to notice an oversight in the script that caused `./tools/nuget.exe` to be run instead of `dotnet restore` (an imported script called `NuGetRestore` instead of `DotNetCoreRestore`). Since the Nuget binary in the tools folder (v5.11, added by the bootstrap) doesn't support .NET 6.0 projects yet it makes sense it would fail.
